### DISCUSSION TITLE
Metadata for the 10k dataset

### DISCUSSION THREAD

**Discussion by hutec** [bdd100k/bdd100k#38] [Q&A] [answered]
Hi,

thanks for a very nice dataset and this repo!

I'm wondering if the metadata attributes (`weather`, `scene` and `timeofday`) are available for the 10k datasets?

**Comment by fyu** [comment, marked as answer]
It is not available to all the images for the segmentation dataset yet.

**Reply by hutec** [reply]
Thanks for the reply. Is there a plan to release it for all images of the segmentation dataset?

**Reply by fyu** [reply]
yes, we are revising the labels on our instance segmentation dataset.

**Reply by pavanteja295** [reply]
Hello, any tentative date as when you can release the metadata for the 10k dataset? Thanks in advance

**Reply by sh-jj** [reply]
Hi, such a dataset with rich attribute information is really helpful!
Ummm, my question is how to download the meta-information (for IMAGE 100K data). I have searched a long time, but I still do not find which file I need to download on the download page (https://bdd-data.berkeley.edu/portal.html#download). 😂
Could you please tell me which file contains the meta-information (weather, time, scene)?
Thanks!

**Reply by fyu** [reply]
The attributes can be downloaded by the `Label` button https://doc.bdd100k.com/download.html#labels

**Reply by sh-jj** [reply]
> The attributes can be downloaded by the `Label` button https://doc.bdd100k.com/download.html#labels

Thanks again！ It's really helpful for my research.

**Reply by SnowdenLee** [reply]
Hi @fyu,
is there any update about releasing the meta information for 10K dataset?  Thanks a lot in advance!

**Reply by Nimra28** [reply]
Can any one tell me the PC requirements to work on this dataset? @fyu

**Comment by HaoLin97** [comment]
@fyu Is there any update on this?

**Comment by Nimra28** [comment]
Can any one tell me the PC requirements to work on this dataset? @fyu

**Comment by okaynils** [comment]
### Using available Metadata for the 10k Dataset

As @fyu has mentioned, metadata is not directly available for the 10k dataset. 

I got curious since I am currently taking a deep learning class in which I train and compare semantic segmentation models to take a closer look at the impact types of scenes in self-driving environments have. 

It looks to me like the 10k dataset contains sampled images from the 100k dataset so when I tried to match the image filename references in the `det_train.json` and `det_val.json` to the 10k images filenames I found that there is some overlap. Of the 10k images `3426` have an annotated reference in either the `det_train.json` or `det_val.json`. All the matches reference to images inside the `train`-partition of the 10k dataset. So not all 10k images are annotated thus only the `3426` of `8000` train-images can be accessed with existing attributes (`"scene"`, `"weather"` and `"timeofday"`).

Note: Again, since this **overlap only has references to the train-partition**, anyone interested in using the metadata would need to partition the dataset themselves.

You can find the `det_train.json` and `det_val.json` available to download at https://dl.cv.ethz.ch/bdd100k/data/.

**Reply by wang1528186571** [reply]
Has the conflict been resolved in the json in the link above? If not, could you provide a resolved version?

**Reply by okaynils** [reply]
> Has the conflict been resolved in the json in the link above? If not, could you provide a resolved version?

@wang1528186571 I do not have a resolved .json but I have made a PyTorch Dataset class that only loads in samples that have scene metadata: https://github.com/okaynils/fhnw-ds-dlbs/blob/main/data/bdd100k_dataset.py
